### PR TITLE
fix: DBTP-907 Platform helper generate version file

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**

### CODE REVIEW

**Review by tony-griffin — APPROVED** [review]

File: dbt_platform_helper/utils/versioning.py [hunk 1]
```diff
@@ -143,6 +144,14 @@ def validate_platform_helper_file_version(template_file_path: str):
     )
 
 
+def generate_platform_helper_version_file(directory="."):
+    base_path = Path(directory)
+
+    # add .platform-helper-version file
```
Comment (WillGibson) [inline dbt_platform_helper/utils/versioning.py:150]: I don't think this comment tells us anything the code doesn't already.
```suggestion
```

File: dbt_platform_helper/commands/generate.py [hunk 2]
```diff
@@ -17,6 +18,7 @@ def generate(ctx: click.Context):
     Wraps pipeline generate and make-addons.
     """
 
+    generate_platform_helper_version_file()
```
[the remark this answers is not in the document]
Comment (samuele-mattiuzzo) [reply dbt_platform_helper/commands/generate.py:21]: @WillGibson tests added! Also learned how to patch a function's set of default arguments, which was quite cool

File: .platform-helper-version
Comment (WillGibson) [inline .platform-helper-version:1]: This should only be committed for the `*-deploy` codebases.

File: tests/platform_helper/test_command_generate.py [hunk 3]
```diff
@@ -35,3 +36,43 @@ def test_platform_helper_generate_shows_a_warning_when_version_is_different_than
...
+    assert version_file_path.read_text() == contents
+
+
+@patch(
+    "dbt_platform_helper.utils.versioning.get_app_versions",
+    new=Mock(return_value=[(1, 0, 0), (1, 0, 0)]),
+)
+@patch("dbt_platform_helper.commands.generate.make_addons", new=Mock(return_value=None))
+@patch("dbt_platform_helper.commands.generate.pipeline_generate", new=Mock(return_value=None))
+def test_platform_helper_generate_does_not_override_version_file_if(tmp_path):
```
Comment (WillGibson) [inline tests/platform_helper/test_command_generate.py:66]:
```suggestion
def test_platform_helper_generate_does_not_override_version_file_if_exists(tmp_path):
```

Comment (samuele-mattiuzzo) [reply tests/platform_helper/test_command_generate.py:66]: silly me. I even saw it and forgot to fix it.

File: .platform-helper-version
Comment (samuele-mattiuzzo) [reply .platform-helper-version:1]: gone 👍🏻

**Review by WillGibson — APPROVED** [review]

**Pull request merged**